Framework/Algorithms/src/CorelliPowderCalibrationDatabase.cpp

```diff
@@ -0,0 +1,718 @@
+  std::string datestamp = year + month + day;
+
+  return datestamp;
+}
+
+//-----------------------------------------------------------------------------
+/**
+ * @brief Compose a standard full path of a component CSV file
+ * @param componentname
+ * @param directory
+ * @return
+ */
+std::string CorelliPowderCalibrationDatabase::corelliComponentDatabaseName(
+    const std::string &componentname, const std::string &directory) {
+  std::string basename = componentname + ".csv";
+  std::string filename = joinPath(directory, basename);
+
+  return filename;
+}
+
+//-----------------------------------------------------------------------------
+/**
+ * @brief Compose a standard full path of a calibration table CSV file
+ * @param datestamp
+ * @param directory
+ * @return
+ */
+std::string CorelliPowderCalibrationDatabase::corelliCalibrationDatabaseName(
+    const std::string &datestamp, const std::string &directory) {
+  std::string basename = datestamp + ".csv";
+  std::string filename = joinPath(directory, basename);
+  return filename;
+}
+
+//-----------------------------------------------------------------------------
+/**
+ * @brief Check whether a file does exist
+ * @param filepath
+ * @return
+ */
+bool CorelliPowderCalibrationDatabase::isFileExist(
+    const std::string &filepath) {
+
+  // TODO - replace by std::filesystem::exists(filename) until C++17 is properly
+  // supported
+  return boost::filesystem::exists(filepath);
+}
+
+//-----------------------------------------------------------------------------
+/**
+ * @brief Join directory and base file name for a full path
+ * @param directory
+ * @param basename
+ * @return
+ */
+std::string
+CorelliPowderCalibrationDatabase::joinPath(const std::string directory,
+                                           const std::string basename) {
+  boost::filesystem::path dir(directory);
+  boost::filesystem::path file(basename);
+  boost::filesystem::path fullpath = dir / file;
+
+  return fullpath.string();
+}
+
+/**
+ * @brief Set up component map
+ * @param componentnames
+ * @param compmap
+ */
+void CorelliPowderCalibrationDatabase::setComponentMap(
+    std::vector<std::string> componentnames,
+    std::map<std::string, DataObjects::TableWorkspace_sptr> &compmap) {
+  // Add entries
+  for (auto compname : componentnames)
+    compmap[compname] = nullptr;
+}
+
+/**
+ * @brief Retrieve the compoments including banks, source and sample names from
+ * an workspace
+ * @param ws : Any workspace containing instrument
+ * @return  : vector including of all the compoments in the order as source,
+ * sample, bank1, bank2, ...
+ */
+std::vector<std::string>
+CorelliPowderCalibrationDatabase::retrieveInstrumentComponents(
+    MatrixWorkspace_sptr ws) {
+  // Get access to instrument information
+  const auto &component_info = ws->componentInfo();
+
+  // Init output
+  std::vector<std::string> componentnames = {"source", "sample"};
```

Comment:
It turns out that in the CORELLI instrument definition file, [the component name for the source](https://github.com/mantidproject/mantid/blob/master/instrument/CORELLI_Definition.xml#L15) is "moderator", and [the component name for the sample](https://github.com/mantidproject/mantid/blob/master/instrument/CORELLI_Definition.xml#L30) is "sample-position". Can you change these names, please?

Reply:
so be it:
- source -> moderator
- sample -> sample-position